### ISSUE TITLE
error Panicking because nodes appear to be being changed every time we run queries. This would cause the site to recompile infinitely. 

### ISSUE THREAD

**Issue by baloghrebecca** [gatsbyjs/gatsby#28625]
## Description

Following error is thrown when running **gatsby develop**: 

error Panicking because nodes appear to be being changed every time we run queries. This would cause the site to recompile infinitely. 
Check custom resolvers to see if they are unconditionally creating or mutating nodes on every query. 
This may happen if they create nodes with a field that is different every time, such as a timestamp or unique id.

![Bildschirmfoto 2020-12-15 um 13 44 37](https://user-images.githubusercontent.com/55947268/102216609-b166db00-3edb-11eb-98e1-28c5d9dce9d0.png)

I'm grateful for any help!

### Steps to reproduce

I'd love to but I'm not sure at all what could be causing it. But here is my link to the [github repo](https://github.com/baloghrebecca/reproduction).

1. Just run gatsby develop
(sometimes it also gets stuck at "source and transform nodes" for several minutes, another issue that I need to solve)
2. See error thrown in the logs

### Expected result

The development environment should start

### Actual result

Throws the following error: 
error Panicking because nodes appear to be being changed every time we run queries. This would cause the site to recompile infinitely. 
Check custom resolvers to see if they are unconditionally creating or mutating nodes on every query. 
This may happen if they create nodes with a field that is different every time, such as a timestamp or unique id.

### Environment

  System:
    OS: macOS 10.15.7
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.5.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.9 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 87.0.4280.88
    Edge: 87.0.664.60
    Firefox: 82.0.3
    Safari: 14.0
  npmPackages:
    gatsby: ^2.28.2 => 2.28.2 
    gatsby-image: ^2.4.12 => 2.4.12 
    gatsby-plugin-anchor-links: ^1.1.1 => 1.1.1 
    gatsby-plugin-favicon: ^3.1.6 => 3.1.6 
    gatsby-plugin-offline: ^3.2.9 => 3.2.9 
    gatsby-plugin-react-helmet: ^3.3.4 => 3.3.4 
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0 
    gatsby-plugin-sass: ^2.3.4 => 2.3.4 
    gatsby-plugin-sharp: ^2.6.18 => 2.6.18 
    gatsby-plugin-smoothscroll: ^1.1.0 => 1.1.0 
    gatsby-plugin-snipcart: ^1.1.2 => 1.1.2 
    gatsby-plugin-snipcart-advanced: ^1.0.0 => 1.0.0 
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4 
    gatsby-remark-images: ^3.3.10 => 3.3.10 
    gatsby-source-filesystem: ^2.0.28 => 2.0.28 
    gatsby-source-strapi: 0.0.12 => 0.0.12 
    gatsby-source-stripe: ^3.1.1 => 3.1.1 
    gatsby-transformer-remark: ^2.8.17 => 2.8.17 
    gatsby-transformer-sharp: ^2.5.10 => 2.5.10 
  npmGlobalPackages:
    gatsby-cli: 2.15.1

**Comment by vladar** [comment]
Thank you for opening this, @baloghrebecca 

I did a quick investigation and looks like it is caused by `gatsby-source-strapi` plugin. You have many files and somehow it seems to be downloading them in parallel with query running (should do it **before** query running - in `source nodes` step).

Once I've upgraded the plugin version to `gatsby-source-strapi@1.0.0-alpha.0` (latest) I can't reproduce this problem - they've probably fixed the issue that was causing this. Now all media files are downloaded during `source nodes` and then queries are executed normally.

We're marking this issue as answered and closing it for now but please feel free to reopen this and comment if you would like to continue this discussion. We hope we managed to help and thank you for using Gatsby! 💜

**Comment by danm** [comment]
I'm getting this error too whilst using Docz. `gatsby-source-strapi` isn't included in my project.

Because I'm abstracting Gatsby by using Docz I'm unsure of the internals or what creates this error. Are there any obvious things to check first in what could be causing this before I try to recreate in a minimal repo?

**Comment by ordishc** [comment]
I'm also getting this error and I do have `"gatsby-source-strapi": "^0.0.12",` installed.

**Comment by AshkanArabim** [comment]
(may not be directly related, but...)
In my case, this issue was caused by this section (which I copied from [this tutorial](https://www.gatsbyjs.com/docs/how-to/sourcing-data/sourcing-from-the-filesystem/)) in my `gatsby-config.ts` file (yes; I'm using typescript):
```
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `src`,
        path: `${__dirname}/src/`,
      },
    },
```
Gatsby, when used with TS, generates a `gatsby-types.d.ts` under `<workdir>/src/` as part of every compilation. What happens in this case, is that since `gatsby-types.d.ts` is updated after every compile, Gatsby keeps detecting it as a source code change and triggers a recompile, which causes `gatsby-types.d.ts` to change again and so on.

I wanted to use `gatsby-source-filesystem` to render my markdown blog entries, so all I had to do was to change `${__dirname}/src/` to `${__dirname}/src/content/` where I have my `.md` files.
